NLSY/nlsy_impute.R

```diff
@@ -0,0 +1,160 @@
+test$hisp <- ifelse(test$hispanic == "Yes", 1, 0)
+test$ret_sav <- ifelse(test$savings == "YES", 1, 0)
+test$hown <- ifelse(test$home == "OWNS OR IS BUYING; LAND CONTRACT", 1, 0)
+test$both_pars <- ifelse(test$both_parents == "Yes", 1, 0)
+test$par_dec <- ifelse(test$par1_deceased == "Yes" | test$par2_deceased == "Yes", 1, 0) # Either parent deceased
+test$mom_educ_hs <- ifelse(test$mom_education == "12TH GRADE" | test$mom_education == "1ST YEAR COLLEGE" | test$mom_education == "2ND YEAR COLLEGE" |  test$mom_education == "3RD YEAR COLLEGE" |  test$mom_education == "4TH YEAR COLLEGE" |  test$mom_education == "5TH YEAR COLLEGE" |  test$mom_education == "6TH YEAR COLLEGE" |  test$mom_education == "7TH YEAR COLLEGE" |  test$mom_education == "8TH YEAR COLLEGE", 1, 0)
+test$dad_educ_hs <- ifelse(test$dad_education == "12TH GRADE" | test$dad_education == "1ST YEAR COLLEGE" | test$dad_education == "2ND YEAR COLLEGE" |  test$dad_education == "3RD YEAR COLLEGE" |  test$mom_education == "4TH YEAR COLLEGE" |  test$mom_education == "5TH YEAR COLLEGE" |  test$mom_education == "6TH YEAR COLLEGE" |  test$mom_education == "7TH YEAR COLLEGE" |  test$mom_education == "8TH YEAR COLLEGE", 1, 0)
+test$race_w <- ifelse(test$race == "White", 1, 0)
+test$race_b <- ifelse(test$race == "Black or African American", 1, 0)
+
+# Looking for significance with non-missing income and wealth
+non_missing <- test %>%
+    filter(!is.na(pincome),
+           !is.na(pnetworth))
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_w + race_b + mom_educ_hs + dad_educ_hs + ret_sav + hown + both_pars + par_dec, data = non_missing))
+#Non-significant: race_w and par_dec - dropping for now
+
+# Checking for multi-collinearity
+testing_cor <- test %>%
+    select(-id, -hispanic, -race, -mom_education, -dad_education, -savings, -home, -both_parents, -par1_deceased, -par2_deceased, - mom_age_birth, -pincome, -pnetworth, -race_w, -par_dec) %>%
+    drop_na()
+
+M <- cor(testing_cor)
+corrplot(M, type = "upper")
+
+summary(model <- lm(pincome ~ mom_age, data = test)) #Adj R2 = 0.057
+summary(model <- lm(pincome ~ mom_age + hisp, data = test)) #Adj R2 = 0.09
+car::vif(model)
+summary(model <- lm(pincome ~ mom_age + hisp + race_b, data = test)) # #Adj R2 = 0.154
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs, data = test)) #Adj R2 = 0.1767
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs + dad_educ_hs, data = test)) #Adj R2 = 0.1336
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs + dad_educ_hs + ret_sav, data = test)) #Adj R2 = 0.2047 - biggest R2 jump
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs + dad_educ_hs + ret_sav + hown, data = test)) #Adj R2 = 0.2165
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race_b + mom_educ_hs + dad_educ_hs + ret_sav + hown + both_pars, data = test)) #Adj R2 = 0.2178
```

Comment:
I think that it would be good to add interactions between ethnicity and race, as well as between mom's and dad's education.

Reply:
Just flagging a significant amount of dad's education is missing - if we decide to drop it, then that affects the interactions (see my comment above). Otherwise, totally agree on including the interactions!

Reply:
To close the loop on this - the interaction term and the race term were causing the mice package to report a singularity error (likely multicollinearity, which was difficult to pick up given they are both now factor variables). In the final imputation, I have only used the interaction term. Since father education is dropped due to missingness, that interaction is also not created.